lib/msal-browser/src/cache/BrowserCacheManager.ts

```diff
@@ -393,30 +818,44 @@ export class BrowserCacheManager extends CacheManager {
      * @param value
      */
     setItem(key: string, value: string, correlationId: string): void {
-        let tokenKeysV0Count = 0;
-        let accessTokenKeys: Array<string> = [];
+        const tokenKeysCount = new Array(
+            CacheKeys.CREDENTIAL_SCHEMA_VERSION + 1
+        ).fill(0);
+        const accessTokenKeys: Array<string> = [];
         const maxRetries = 20;
         for (let i = 0; i <= maxRetries; i++) {
             try {
                 this.browserStorage.setItem(key, value);
```

Comment:
nit: It's a bit unclear why the setItem call would fail here. IIUC from the context + the code below, if the cache quota is met then tokens are removed starting with the oldest schema for setItem to succeed in the next attempt. It'd be helpful to have the logic/index math here and  below have a few more comments to spell it out a bit.

Reply:
Done